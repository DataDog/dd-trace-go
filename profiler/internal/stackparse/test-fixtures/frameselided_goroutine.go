//+build ignore

// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2016 Datadog, Inc.

package main

import (
	"os"
	"runtime/pprof"
)

func main() {
	wait := make(chan struct{})
	go stackDump(100, wait)
	<-wait
}

func stackDump(remaining int, done chan struct{}) {
	if remaining > 0 {
		stackDump(remaining-1, done)
	} else {
		pprof.Lookup("goroutine").WriteTo(os.Stdout, 2)
		done <- struct{}{}
	}
}
