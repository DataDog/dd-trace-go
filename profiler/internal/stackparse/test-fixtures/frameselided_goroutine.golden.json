{
  "Errors": null,
  "Goroutines": [
    {
      "ID": 18,
      "State": "running",
      "Wait": 0,
      "LockedToThread": false,
      "Stack": [
        {
          "Func": "runtime/pprof.writeGoroutineStacks",
          "File": "/usr/local/Cellar/go/1.15.6/libexec/src/runtime/pprof/pprof.go",
          "Line": 693
        },
        {
          "Func": "runtime/pprof.writeGoroutine",
          "File": "/usr/local/Cellar/go/1.15.6/libexec/src/runtime/pprof/pprof.go",
          "Line": 682
        },
        {
          "Func": "runtime/pprof.(*Profile).WriteTo",
          "File": "/usr/local/Cellar/go/1.15.6/libexec/src/runtime/pprof/pprof.go",
          "Line": 331
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 24
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        },
        {
          "Func": "main.stackDump",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 22
        }
      ],
      "FramesElided": true,
      "CreatedBy": {
        "Func": "main.main",
        "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
        "Line": 16
      }
    },
    {
      "ID": 1,
      "State": "chan receive",
      "Wait": 0,
      "LockedToThread": false,
      "Stack": [
        {
          "Func": "main.main",
          "File": "/Users/felix.geisendoerfer/go/src/github.com/DataDog/dd-trace-go/profiler/internal/stackparse/test-fixtures/frameselided_goroutine.go",
          "Line": 17
        }
      ],
      "FramesElided": false,
      "CreatedBy": null
    }
  ]
}
